src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts

```diff
@@ -0,0 +1,184 @@
+import { TrackInfo, TrackInfoStream } from '@taktik/mux.js'
+import { FfmpegCommand } from 'fluent-ffmpeg'
+import { PassThrough, Readable } from 'stream'
+import { State, StateMachineImpl } from 'typescript-state-machine'
+import { getLogger } from '../../logging/loggers'
+import { Dispatcher } from '../dispatcher'
+import { FlowrFfmpeg } from '../ffmpeg'
+import { IpcStreamer } from '../ipcStreamer'
+import { PlayerError, PlayerErrors } from '../playerError'
+
+enum PipelineStateName {
+  STARTING = 'Starting',
+  RETRIEVING_METADATA = 'Retrieving metadata',
+  RETRIEVED_METADATA = 'Retrieved metadata',
+  PLAYING = 'Playing',
+  KILLED = 'Killed',
+  ERROR = 'Error',
+}
+
+class PipelineState extends State {
+  label: PipelineStateName
+}
+
+const STARTING = new PipelineState(PipelineStateName.STARTING)
+const RETRIEVING_METADATA = new PipelineState(PipelineStateName.RETRIEVING_METADATA)
+const RETRIEVED_METADATA = new PipelineState(PipelineStateName.RETRIEVED_METADATA)
+const PLAYING = new PipelineState(PipelineStateName.PLAYING)
+const KILLED = new PipelineState(PipelineStateName.KILLED)
+const ERROR = new PipelineState(PipelineStateName.ERROR)
+
+const PlayingPipelineStates = { STARTING, RETRIEVING_METADATA, RETRIEVED_METADATA, PLAYING, KILLED, ERROR }
+
+const transitions = {
+  [PipelineStateName.STARTING]: [RETRIEVING_METADATA, ERROR],
+  [PipelineStateName.RETRIEVING_METADATA]: [RETRIEVED_METADATA, KILLED, ERROR],
+  [PipelineStateName.RETRIEVED_METADATA]: [PLAYING, ERROR],
+  [PipelineStateName.PLAYING]: [KILLED, ERROR],
+  [PipelineStateName.KILLED]: [] as [],
```

Comment:
[] as [], this is normal ?

Reply:
:'(

<img width="602" alt="image" src="https://user-images.githubusercontent.com/5877409/193214894-8b937483-8f75-48a5-8733-91d5ac882aa9.png">